### PR TITLE
Fix rust escape double quots in string

### PR DESCRIPTION
**Pull request by MohamedAli00949**
I have fixed unescaping the `"` at string attributes such as I have seen in the #3817 issue.

### Changes

The string scops didn't contain the backslash escape expression which is `BACKSLASH_ESCAPE` so I have put it at every string scope.

Before
image

After
rust-issue

### Checklist
- [x] Add `BACKSLASH_ESCAPE` to string scops to avoid this issue.
- [x] Add markup tests to test the validation of the solution.

### CODE REVIEW

**Comment by MohamedAli00949** [comment]
hi @joshgoebel, this is a simple issue and feel free to review it and give me your pieces of advice.

File: src/languages/rust.js [hunk 1]
```diff
@@ -193,9 +193,19 @@ export default function(hljs) {
...
+              hljs.BACKSLASH_ESCAPE
+            ]
+          },
+          {
+            begin: /b?'\\?(x\w{2}|u\w{4}|U\w{8}|.)'/,
+            contains: [
+              hljs.BACKSLASH_ESCAPE
+            ]
+          },
+        ],
```
Comment (joshgoebel) [inline src/languages/rust.js:208]: This doesn't work... you'd need `end` as well and need to rewrite any specific internals (like the unicode matching) to be sub rules inside the `contains`.  And you can't use match history (`\1`) across begin/end combos... how many `#` characters are used in practice? is there a max limit?

Comment (MohamedAli00949) [reply src/languages/rust.js:208]: I am sorry, I can't understand you.
Do you mean removing this change avoiding complexity or another thing?

Comment (joshgoebel) [reply src/languages/rust.js:208]: Please add test cases for these and you will see that they do not work they way they are currently written.

Comment (MohamedAli00949) [reply src/languages/rust.js:208]: You are right, this change didn't work and the contains at this these variants.
I will remove it and commit changes again.
Thanks in advance

**Comment by joshgoebel** [comment]
Ok, I supposing fixing one of the variants s better than nothing.  Please add a changelog entry.

**Comment by MohamedAli00949** [comment]
@joshgoebel, I have updated the changelogs at CHANGES.md Is that what you expect?

**Comment by joshgoebel** [comment]
Please only include your single change in the log.

**Review by joshgoebel — APPROVED** [review]

**Comment by MohamedAli00949** [comment]
@joshgoebel, Thanks for approving the changes, I hope to merge it if possible.

**Pull request merged**